src/time/BrokenDate.cpp

```diff
@@ -26,6 +26,38 @@ Copyright_License {
 #include "Calendar.hxx"
 
 #include <cassert>
+#include <vector>
+#include <algorithm>
+#include "boost/algorithm/string.hpp"
+
+#ifdef _UNICODE
+typedef std::vector<std::wstring> StringVector;
+#else
+typedef std::vector<std::string> StringVector;
+#endif
+
+BrokenDate::BrokenDate(const TCHAR* date_string) : year(0), month(0), day(0)
+{
+  StringVector strs;
+  boost::split(strs, date_string, boost::is_any_of(_T("/-.")));
+  if (strs.size() == 3) {  // ISO 8601(2003-01-07), american style (2003/01/07)
```

Comment:
Very bad and unreliable way of detecting date formats. Please, no!

Reply:
removed